src/sshfs_mount/sftp_server.cpp

```diff
@@ -508,24 +470,26 @@ int mp::SftpServer::handle_mkdir(sftp_client_message msg)
     const auto filename = sftp_client_message_get_filename(msg);
     if (!validate_path(source_path, filename))
     {
-        mpl::log(
-            mpl::Level::trace, category,
-            fmt::format("{}: cannot validate path \'{}\' against source \'{}\'", __FUNCTION__, filename, source_path));
+        mpl::log(mpl::Level::trace,
+                 category,
+                 fmt::format("{}: cannot validate path '{}' against source '{}'", __FUNCTION__, filename, source_path));
         return reply_perm_denied(msg);
     }
 
     QDir dir(filename);
     if (!dir.mkdir(filename))
     {
-        mpl::log(mpl::Level::trace, category, fmt::format("{}: mkdir failed for \'{}\'", __FUNCTION__, filename));
+        mpl::log(mpl::Level::trace, category, fmt::format("{}: mkdir failed for '{}'", __FUNCTION__, filename));
         return reply_failure(msg);
     }
 
-    QFile file(filename);
-    if (!MP_FILEOPS.setPermissions(file, to_qt_permissions(msg->attr->permissions)))
+    std::error_code err;
+    MP_FILEOPS.permissions(filename, static_cast<fs::perms>(msg->attr->permissions), err);
```

Comment:
Maybe `static_cast<fs::perms>(msg->attr->permissions)` -> `fs::perms(msg->attr->permissions)`, both interpret uint32_t into bitmask, but the second one is cleaner.

Reply:
Agreed

Reply:
Hey guys, sorry I noticed this flying by. Why is a functional cast here preferable to a `static_cast`?

Reply:
I guess `{}` is OK too if no narrowing is involved.

Reply:
Yes, I think `{}` is the cleanest. I understand that a C-style cast is not really recommended and, if this was a conversion between more complex data types i.e. classes, I would also definitely prefer named casts. In this case, since it's just an integer to enum conversion, I initially agreed to the C-style cast since it would simply act like the `static_cast` that it replaced. But now that you mention `{}`, I think that is preferable. Thanks for pointing it out!

Reply:
> I guess {} is OK too if no narrowing is involved.

That is what the item of the guideline said. On the top of that () and {} functional cast are also slightly different in terms of narrowing prevention. 

> Yes, I think {} is the cleanest. I understand that a C-style cast is not really recommended and,
I think you mean functional cast, C-style is the (T) var.

Reply:
Ahaa! And thus we just prevented a narrowing conversion.